sympy/solvers/ode/tests/test_riccati.py

```diff
@@ -0,0 +1,945 @@
+
+
+def test_solve_riccati():
+    """
+    This function tests the computation of rational
+    particular solutions for a Riccati ODE.
+
+    Each test case has 2 values -
+
+    1. eq - Riccati ODE to be solved.
+    2. sol - Expected solution to the equation.
+
+    Some examples have been taken from the paper - "Statistical Investigation of
+    First-Order Algebraic ODEs and their Rational General Solutions" by
+    Georg Grasegger, N. Thieu Vo, Franz Winkler
+
+    https://www3.risc.jku.at/publications/download/risc_5197/RISCReport15-19.pdf
+    """
+    C0 = Dummy('C0')
+    # Type: 1st Order Rational Riccati, dy/dx = a + b*y + c*y**2,
+    # a, b, c are rational functions of x
+
+    tests = [
+    (
+        x**2 - (2*x + 1/x)*f(x) + f(x)**2 + f(x).diff(x),
+        [Eq(f(x), (C0*x + x**3 + 2*x)/(C0 + x**2)), Eq(f(x), x)]
+    ),
+    (
+        f(x)**2 + f(x).diff(x) - (4*x**6 - 8*x**5 + 12*x**4 + 4*x**3 + \
+            7*x**2 - 20*x + 4)/(4*x**4),
+        [Eq(f(x), (2*x**5 - 2*x**4 - x**3 + 4*x**2 + 3*x - 2)/(2*x**4 \
+            - 2*x**2))]
+    ),
+    (
+        -x*f(x)**2 + f(x).diff(x) - 2*f(x)/x,
+        [Eq(f(x), -4*x**2/(C0 + x**4)), Eq(f(x), 0)]
+    ),
+    (
+        -f(x)**2 + f(x).diff(x) + (15*x**2 - 20*x + 7)/((x - 1)**2*(2*x \
+            - 1)**2),
+        [Eq(f(x), (9*C0*x - 6*C0 - 15*x**5 + 60*x**4 - 94*x**3 + 72*x**2 \
+        - 30*x + 6)/(6*C0*x**2 - 9*C0*x + 3*C0 + 6*x**6 - 29*x**5 + \
+        57*x**4 - 58*x**3 + 30*x**2 - 6*x)), Eq(f(x), (3*x - 2)/(2*x**2 \
+        - 3*x + 1))]
+    ),
+    (
+        9*x**2/4 - f(x)**2 + f(x).diff(x) - S(21)/2,
+        [Eq(f(x), 3*x/2 - (3*x**2 - 1)/(x*(x**2 - 1)))]
+    ),
+    (
+        f(x)**2 + f(x).diff(x) - 15/(4*x**2),
+        [Eq(f(x), (-3*C0 + 5*x**4)/(2*C0*x + 2*x**5)), Eq(f(x), -3/(2*x))]
+    ),
+    (
+        3*f(x)**2 + f(x).diff(x) - 2/x**2,
+        [Eq(f(x), (-2*C0 + 3*x**5)/(3*C0*x + 3*x**6)), Eq(f(x), -2/(3*x))]
+    ),
+    (
+        f(x).diff(x) - 2*I*(f(x)**2 + 1)/x,
+        [Eq(f(x), (-I*C0 + I*x**4)/(C0 + x**4)), Eq(f(x), -I)]
+    ),
+    (
+        f(x).diff(x) - f(x)**2/x + 1/x,
+        [Eq(f(x), (C0 - x**2)/(C0 + x**2)), Eq(f(x), 1)]
+    ),
+    (
+        f(x)**2 + f(x).diff(x) - f(x)/x,
+        [Eq(f(x), 2*x/(C0 + x**2)), Eq(f(x), 0)]
+    ),
+    (
+        -x**2 - (2*x + 1/x)*f(x) - f(x)**2 + f(x).diff(x),
+        [Eq(f(x), (-C0*x - x**3 - 2*x)/(C0 + x**2)), Eq(f(x), -x)]
+    ),
+    (
+        f(x)**2 + f(x).diff(x) + 4*f(x)/x + 2/x**2,
+        [Eq(f(x), (-2*C0 - x)/(C0*x + x**2))]
+    ),
+    (
+        2*x**2*f(x).diff(x) - x*(4*f(x) + f(x).diff(x) - 4) + (f(x) - 1)*f(x),
+        [Eq(f(x), (C0 + 2*x**2)/(C0 + x))]
+    ),
+    (
+        x**4*f(x).diff(x) + x**2 - x*(2*f(x)**2 + f(x).diff(x)) + f(x),
+        [Eq(f(x), (C0*x**2 + x)/(C0 + x**2)), Eq(f(x), x**2)]
+    ),
+    ( # Regression Test: See https://github.com/sympy/sympy/pull/21459#issuecomment-852212302
```

Comment:
Rather than linking to a comment it is better to explain why this case is significant compared to the other test examples (the linked comment doesn't offer any further explanation).

In general in this PR I find it hard to understand what the significance of any of the test cases is. They all just look like random examples to me which makes me think:

1. Do we need this many? If almost all of them are basically just testing the same thing then it would be better to have fewer tests.
2. Are there reasons why you added all of these particular examples?
3. If there are reasons for each of these then it would be good to have comments that explain what the examples are or to categorise them somehow.

Reply:
Some of the tests are from a paper written by the same author which had examples from various sources. 2 or 3 of them are from books that Nijso suggested and the last few of them are the ones that were failing previously because of the bug. Maybe I could remove some of the previously failing examples since they are testing the same case.

Reply:
The main thing is to add some comments explaining what these are testing.